### DISCUSSION TITLE
Improving input history

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#2103] [Ideas]
I defined a command `rfind` on `DirSheet` which let's me type a regex for a file name, and then it shows all files in that folder recursively whose names match that expression. One annoyance though is that the input prompt for this command shares history with all the other commands that take a regex, which for me is mostly commands that filter by column. This is annoying because the file names are completely different from the column names, so I always have to press `Up` several times to scroll past the irrelevant items to reach the relevant one.

This makes me think of two possible improvements:

1. Allow defining separate input histories. I see now that the items in `input_history.jsonl` look like `{"type": "regex", "input": "foo"}`. They could look like `{"type": "regex", "input": "foo", famlily: "file"}`, and each command will specify a `family` which could be `column`, `row` or `file`. (We can talk about the default behavior if a command doesn't specify a family if you like this idea.)

2. It would be cool if there was a way, when giving input, to say "show me the last 10 inputs and let me choose by pressing a key in 0-9". This would be a lot nicer than scrolling.

**Comment by saulpw** [comment]
For 1, use `input(..., type="rfind")` to keep your rfind history separate from other input history.

**Reply by cool-RR** [reply]
Awesome, thank you!

**Comment by frosencrantz** [comment]
@cool-RR your 2nd bullet looks similar to https://github.com/saulpw/visidata/issues/2081
